xExtension-ImageProxy/extension.php

```diff
@@ -1,45 +1,95 @@
+			// Legacy
+			if (!is_null(FreshRSS_Context::$user_conf->image_proxy_force)) {
+				FreshRSS_Context::$user_conf->image_proxy_scheme_https = FreshRSS_Context::$user_conf->image_proxy_force;
+				FreshRSS_Context::$user_conf->image_proxy_force = null;  // Minz -> unset
+			}
+			$save = true;
+		}
+		if (is_null(FreshRSS_Context::$user_conf->image_proxy_scheme_default)) {
+			FreshRSS_Context::$user_conf->image_proxy_scheme_default = self::SCHEME_DEFAULT;
+			$save = true;
+		}
+		if (is_null(FreshRSS_Context::$user_conf->image_proxy_scheme_include)) {
+			FreshRSS_Context::$user_conf->image_proxy_scheme_include = self::SCHEME_INCLUDE;
+			$save = true;
+		}
+		if (is_null(FreshRSS_Context::$user_conf->image_proxy_url_encode)) {
+			FreshRSS_Context::$user_conf->image_proxy_url_encode = self::URL_ENCODE;
+			$save = true;
+		}
+		if ($save) {
+			FreshRSS_Context::$user_conf->save();
 		}
 	}
 
-	public static $proxy_url = 'https://images.weserv.nl/?url=';
-
 	public function handleConfigureAction() {
 		$this->registerTranslates();
 
 		if (Minz_Request::isPost()) {
-			FreshRSS_Context::$user_conf->image_proxy_url = Minz_Request::param('image_proxy_url', '');
-			FreshRSS_Context::$user_conf->image_proxy_force = Minz_Request::param('image_proxy_force', '');
+			FreshRSS_Context::$user_conf->image_proxy_url = Minz_Request::param('image_proxy_url', self::PROXY_URL);
+			FreshRSS_Context::$user_conf->image_proxy_scheme_http = Minz_Request::param('image_proxy_scheme_http', '');
+			FreshRSS_Context::$user_conf->image_proxy_scheme_https = Minz_Request::param('image_proxy_scheme_https', '');
+			FreshRSS_Context::$user_conf->image_proxy_scheme_default = Minz_Request::param('image_proxy_scheme_default', self::SCHEME_DEFAULT);
+			FreshRSS_Context::$user_conf->image_proxy_scheme_include = Minz_Request::param('image_proxy_scheme_include', '');
+			FreshRSS_Context::$user_conf->image_proxy_url_encode = Minz_Request::param('image_proxy_url_encode', '');
 			FreshRSS_Context::$user_conf->save();
 		}
 	}
 
 	public static function getProxyImageUri($url) {
 		$parsed_url = parse_url($url);
-		if (isset($parsed_url['scheme']) && $parsed_url['scheme'] === 'http') {
-			$url = self::$proxy_url . rawurlencode(substr($url, strlen('http://')));
+		$scheme = isset($parsed_url['scheme']) ? $parsed_url['scheme'] : null;
+		if ($scheme === 'http') {
+			if (!FreshRSS_Context::$user_conf->image_proxy_scheme_http) return $url;
+			if (!FreshRSS_Context::$user_conf->image_proxy_scheme_include) {
+				$url = substr($url, 7);  // http://
+			}
+		}
+		else if ($scheme === 'https') {
+			if (!FreshRSS_Context::$user_conf->image_proxy_scheme_https) return $url;
+			if (!FreshRSS_Context::$user_conf->image_proxy_scheme_include) {
+				$url = substr($url, 8);  // https://
+			}
+		}
+		else if (empty($scheme)) {
+			if (substr(FreshRSS_Context::$user_conf->image_proxy_scheme_default,0, 4) !== 'http') return $url;
```

Comment:
Does this break with the purposeful behavior described in the comment on line 36 and executed on line 39? I'd suggest making it an option if you consider it undesirable. I didn't stick it in just for fun. ;-)

See https://www.paulirish.com/2010/the-protocol-relative-url/ for what it's talking about. A URL that looks like `//websbestgifs.net/kittyonadolphin.gif`.

Reply:
You're right! Sorry, I did not mean to break this behavior. I believe https://github.com/FreshRSS/Extensions/pull/77/commits/cc24497c5c42b79131b770e9c0b755fc7c906d18 fixes this and maintains backward compatibility.